pkg/cli/demo.go

```diff
@@ -1029,3 +997,90 @@ func runDemo(cmd *cobra.Command, gen workload.Generator) (err error) {
 
 	return runClient(cmd, conn)
 }
+
+// getServerURL generates a connection URL for the given server.
+func (c *transientCluster) getServerURL(
+	nodeID roachpb.NodeID, gen workload.Generator, includeAppName bool,
+) (url.URL, error) {
+	return url.URL{}, nil
+}
+
+// sockForServer generates the metadata for a unix socket for the given node.
+// For example, node 1 gets socket /tmpdemodir/.s.PGSQL.26267,
+// node 2 gets socket /tmpdemodir/.s.PGSQL.26268, etc.
+func (c *transientCluster) sockForServer(nodeID roachpb.NodeID) unixSocketDetails {
+	if !c.useSockets {
+		return unixSocketDetails{}
+	}
+	defaultPort, _ := strconv.Atoi(base.DefaultPort)
+	return unixSocketDetails{
+		socketDir:  c.demoDir,
+		portNumber: defaultPort + int(nodeID) - 1,
```

Comment:
is this going to cause a problem if i have 2 cockroach demo shells open at the same time?